streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -488,10 +488,13 @@ private void runLoop() {
     long runOnce(long recordsProcessedBeforeCommit) {
         timerStartedMs = time.milliseconds();
 
-        // try to fetch some records if necessary
-        final ConsumerRecords<byte[], byte[]> records = pollRequests();
+        ConsumerRecords<byte[], byte[]> records;
 
         if (state == State.PARTITIONS_ASSIGNED) {
+            // try to fetch some records with zero poll millis
+            // to unblock the restoration as soon as possible
+            records = pollRequests(0L);
```

Comment:
since the restore loop will now be very tight, we should also change the log level in  `StoreChangelogReader.completed()` to `trace` to avoid unnecessary logging at the debug level

Reply:
we already changed it in trunk and 1.0 [here](https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java#L197) so I don't think this should be too controversial